pyflow/RCDT2/Nodes/core.py

```diff
@@ -0,0 +1,183 @@
+            )
+            return
+        logger.info("Response was succesfull.")
+        self.set_response_to_pins(response)
+        self.exec_out.call()
+
+    def create_data_pins(self, service_part: str) -> None:
+        msg: type = getattr(self.service_type, service_part)
+        fields_and_field_types: dict[str, str] = msg.get_fields_and_field_types()
+        for field, field_type in fields_and_field_types.items():
+            pin_type = get_pin_type(field_type)
+            if msg.__name__ == self.service_type.__name__ + "_Request":
+                pin = self.createInputPin(field, pin_type)
+                self.pin_manager.add_input_pin(field, pin)
+            elif msg.__name__ == self.service_type.__name__ + "_Response":
+                pin = self.createOutputPin(field, pin_type)
+                self.pin_manager.add_output_pin(field, pin)
+            else:
+                logger.warning("Message is not a Request or a Response. Skip.")
+
+    def create_request_from_pins(self) -> object:
+        request = self.service_type.Request()
+        for pin_name in self.pin_manager.get_input_pin_names():
+            data = self.pin_manager.get_data(pin_name)
+            setattr(request, pin_name, data)
+        return request
+
+    def set_response_to_pins(self, response: object) -> None:
+        for pin_name in self.pin_manager.get_output_pin_names():
+            data = getattr(response, pin_name)
+            self.pin_manager.set_data(pin_name, data)
+
+    def call_service(self, request: object) -> object:
+        logger.info("Calling service...")
+        response = self.service_type.Response()
+        response.success = False
+        if not self.client.wait_for_service(3):
+            logger.error("Service not available. Exit.")
+            return response
+
+        future = self.client.call_async(request)
+        timeout = 20
+        waited = 0
+        while not future.done():
+            if waited >= timeout:
+                future.cancel()
+                logger.error("No response from service. Exit.")
+                return response
+            sleep(1)
+            waited += 1
+        return future.result()
+
+    @staticmethod
+    def category() -> None:
+        return "Services"
+
+
+def get_pin_type(field_type: str) -> str:
+    """
+    Connect ROS built-in-types to default pins.
+    https://docs.ros.org/en/jazzy/Concepts/Basic/About-Interfaces.html#field-types
+    Otherwise field_type is a ROS message, of which a pin should be made automatically.
+    """
+    if field_type == "boolean":
+        return "BoolPin"
+    if field_type in ["string", "wstring"]:
+        return "StringPin"
+    if field_type in ["float32", "float64"]:
+        return "FloatPin"
+    if field_type in [
+        "int8",
+        "uint8",
+        "int16",
+        "uint16",
+        "int32",
+        "uint32",
+        "int64",
+        "uint64",
+    ]:
+        return "IntPin"
+    return field_type
```

Comment:
ipv deze if statements met tussentijdse returns (wat je eigenlijk niet wil) kan dit ook
```suggestion
field_type_to_pin_type: dict[str, str] = {
    "boolean": "BoolPin",
    "string": "StringPin",
    "wstring": "StringPin",
    "float32": "FloatPin",
    "float64": "FloatPin",
    "int8": "IntPin",
    "uint8": "IntPin",
    "int16": "IntPin",
    "uint16": "IntPin",
    "int32": "IntPin",
    "uint32": "IntPin",
    "int64": "IntPin",
    "uint64": "IntPin",
}

def get_pin_type(field_type: str) -> str:
    """
    Connect ROS built-in-types to default pins.
    https://docs.ros.org/en/jazzy/Concepts/Basic/About-Interfaces.html#field-types
    Otherwise field_type is a ROS message, of which a pin should be made automatically.
    """
    try:
        pin_type = field_type_to_pin_type[field_type]
    except KeyError:
        pin_type = field_type
    return pin_type
```